mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAOptimizer.scala

```diff
@@ -446,14 +445,14 @@ final class OnlineLDAOptimizer extends LDAOptimizer with Logging {
   override private[clustering] def next(): OnlineLDAOptimizer = {
     val batch = docs.sample(withReplacement = sampleWithReplacement, miniBatchFraction,
       randomGenerator.nextLong())
-    if (batch.isEmpty()) return this
```

Comment:
We still need this, right?

Reply:
I believe, it's redundant now. Anyway, `submitMiniBatch` counts the documents.

Reply:
If that's the case, you should remove the comments below. 
"The methods assumes no empty documents are submitted."

Reply:
The method assumes no empty documents are submitted. But it's perfectly fine to submit an empty RDD. 

And I need to fix a typo.

Reply:
I would prefer to leave the check here
1. performance-wise, sending an empty rdd to go through submitMiniBatch does not make much sense.
2. it will make submitMiniBatch more simple if no empty RDD will be sent, especially when we're filtering empty documents.

Reply:
So technically, you suggest to move the check 

```
if (batchSize == 0) {
      logWarning("No non-empty documents were submitted in the batch.")
      // Therefore, there is no need to update any of the model parameters
      return this
    }
```

to `next()`? Sounds reasonable.

Reply:
But performance wise... Currently we have a single pass. But if we move the check to `next`, `isEmpty` being an action will trigger the materialization of `batch` and then `batch` will be materialized once again when `treeAggregate` is called in `submitMiniBatch`. 

Right?

Reply:
I think isEmpty() is optimized to avoid a full materialization.

Reply:
If still you want to remove the line, would you please add a unit test to ensure `submitMiniBatch` can handle empty rdd? Thanks.

Reply:
I've added the test. Thank you.